### ISSUE TITLE
Make `torch.hypot` accept scalars

### ISSUE THREAD

**Issue by alexshtf** [pytorch/pytorch#167567]
### 🚀 The feature, motivation and pitch

Currently it requires both arguments to be tensors. So implementing somehting like sqrt(u^2 + x^2), where u is x a tensor requires inefficient allocation of tensors. 

Consequently, something like a "soft-plus" in the form $x \to x + \sqrt{a^2 + x^2}$ or normalization to $[-1, 1]$ via $x / \sqrt{a^2 + x^2}$ also become inefficient.

I  hope it's an extremely small change that is easy to do, and can help in some scenarios I described above.

### Alternatives

1. Writing it myself in Triton and implementing the backward pass.
2. Doing the inefficient way `torch.sqrt(u ** 2 + torch.square(x))`, or `torch.hypot(torch.full_like(u**2, x), x)`
3. Writing the above and using torch.compile, which takes quite a long time to compile the first time the function is invoked.

**Comment by isuruf** [comment]
You can do `torch.hypot(torch.tensor(u), x)` where `torch.tensor(u)` is a scalar tensor and does a small constant allocation.

The way to add it would be to make a `aten.hypot.Scalar` aten operator and implement it.

**Comment by alexshtf** [comment]
It becomes more complicated when devises and dtypes are involved, i.e, `torch.hypot(torch.tensor(u, dtype=x.dtype, device=x.device), x)`. Of course, possible, but ugly.

Moreover, won't we pay an overhead for "uploading" the scalar tensor to the GPU every time we invoke the function? I don't know how PyTorch itself is implemented, but to the best of my knowledge in general adding scalar arguments to CUDA kernels is "free". 

Of course, I don't know how easy it is to support "truly" scalar arguments in PyTorch that are both allocation free, and really scalar under the hood. So I don't know how easy it is. It may not be worth it for such a non-common function.

**Comment by alexshtf** [comment]
Well, you're a PyTorch developer. You decide what's best :)

Personally, I would hope to avoid allocating tensors altogether, but if there is no serious overhead for this one tensor allocation then it doesn't really matter, does it?
